packages/std/src/math.rs

```diff
@@ -486,7 +544,74 @@ mod test {
     #[test]
     fn decimal_add() {
         let value = Decimal::one() + Decimal::percent(50); // 1.5
-        assert_eq!(value.0.u128(), DECIMAL_FRACTIONAL.u128() * 3 / 2);
+        assert_eq!(value.0, DECIMAL_FRACTIONAL * 3 / 2);
+    }
+
+    #[test]
+    fn decimal_to_string() {
+        // Integers
+        assert_eq!(Decimal::zero().to_string(), "0");
+        assert_eq!(Decimal::one().to_string(), "1");
+        assert_eq!(Decimal::percent(500).to_string(), "5");
+
+        // Decimals
+        assert_eq!(Decimal::percent(125).to_string(), "1.25");
+        assert_eq!(Decimal::percent(42638).to_string(), "426.38");
+        assert_eq!(Decimal::percent(1).to_string(), "0.01");
+        assert_eq!(Decimal::permille(987).to_string(), "0.987");
+
+        assert_eq!(Decimal(1).to_string(), "0.000000000000000001");
+        assert_eq!(Decimal(10).to_string(), "0.00000000000000001");
+        assert_eq!(Decimal(100).to_string(), "0.0000000000000001");
+        assert_eq!(Decimal(1000).to_string(), "0.000000000000001");
+        assert_eq!(Decimal(10000).to_string(), "0.00000000000001");
+        assert_eq!(Decimal(100000).to_string(), "0.0000000000001");
+        assert_eq!(Decimal(1000000).to_string(), "0.000000000001");
+        assert_eq!(Decimal(10000000).to_string(), "0.00000000001");
+        assert_eq!(Decimal(100000000).to_string(), "0.0000000001");
+        assert_eq!(Decimal(1000000000).to_string(), "0.000000001");
+        assert_eq!(Decimal(10000000000).to_string(), "0.00000001");
+        assert_eq!(Decimal(100000000000).to_string(), "0.0000001");
+        assert_eq!(Decimal(10000000000000).to_string(), "0.00001");
+        assert_eq!(Decimal(100000000000000).to_string(), "0.0001");
+        assert_eq!(Decimal(1000000000000000).to_string(), "0.001");
+        assert_eq!(Decimal(10000000000000000).to_string(), "0.01");
+        assert_eq!(Decimal(100000000000000000).to_string(), "0.1");
+    }
+
+    #[test]
+    fn decimal_serialize() {
+        assert_eq!(to_vec(&Decimal::zero()).unwrap(), br#""0""#);
+        assert_eq!(to_vec(&Decimal::one()).unwrap(), br#""1""#);
+        assert_eq!(to_vec(&Decimal::percent(8)).unwrap(), br#""0.08""#);
+        assert_eq!(to_vec(&Decimal::percent(87)).unwrap(), br#""0.87""#);
+        assert_eq!(to_vec(&Decimal::percent(876)).unwrap(), br#""8.76""#);
+        assert_eq!(to_vec(&Decimal::percent(8765)).unwrap(), br#""87.65""#);
+    }
+
+    #[test]
+    fn decimal_deserialize() {
```

Comment:
Very nice stuff.

You could add a few tests that it rejects invalid values. But that is not a blocker.